contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -81,9 +83,55 @@ contract LidoOracle is ILidoOracle, AragonApp {
     bytes32 internal constant MIN_REPORTABLE_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.minReportableEpochId");
     /// @dev the max id of reported epochs
     bytes32 internal constant MAX_REPORTED_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.maxReportedEpochId");
+    /// @dev storage for the last completed report and its time
+    bytes32 internal constant LAST_COMPLETED_REPORT_TOTAL_SUPPLY = keccak256("lido.LidoOracle.lastCompletedReportTotalSupply");
+    bytes32 internal constant LAST_COMPLETED_REPORT_TIME = keccak256("lido.LidoOracle.lastCompletedReportTime");
+    bytes32 internal constant LAST_COMPLETED_REPORT_VALIDATORS = keccak256("lido.LidoOracle.lastCompletedReportValidators");
     /// @dev storage for all gathered from reports data
     mapping(uint256 => EpochData) private gatheredEpochData;
 
+    /*
+    If we use APR as a basic reference for increase, and expected range is between 1 and 10% APR.
+    So the granularity step for yearly APR is 1%
+    daily_granularity = 0.01 / 365 = 2.74e-05
+    Currently used basisPoints (1e-4) is unable to express such microscopic ratio.
+    So we should take another popular metric to express this - PPM or 1e-6
+    daily_granularity_in_ppm = 2.74e-05 / 1e-6 = 27 (pretty good)
+    with 100% buffer for future spikes it will be 50
+
+    may be changed by Gov
+    */
+    uint128 public ALLOWED_BEACON_BALANCE_INCREASE = 50;  // PPM
+
+    /*
+    When slashing happens, the balance may decrease at a much faster pace.
+    On the start the system will have about 100 validators and we can guess one of them can be slashed for any reason.
+    1/100 = 10000 PPM
+
+    may be changed by Gov
+    */
+    uint128 public ALLOWED_BEACON_BALANCE_DECREASE = 10000;  // PPM
+    uint256 public constant DAY = 60 * 60 * 24;
+    /*
+    Based on this analytics https://etherscan.io/address/0xae7ab96520de3a18e5e111b5eaab095312d7fe84#analytics
+    The max number of daily deposited validators is about 2000 ETH (19 Dec)
+    Divide it by DEPOSIT_SIZE and get the number of daily activated validators 2000 / 32 = 62.5
+    And we leave 50-percent buffer for possible short-time bursts ~= 100
+    */
+    uint128 public ALLOWED_BEACON_VALIDATORS_DAILY_INCREASE = 100;  // validators
+
+    function set_ALLOWED_BEACON_BALANCE_INCREASE(uint128 _ALLOWED_BEACON_BALANCE_INCREASE) external auth(SET_REPORT_LIMITS) {
```

Comment:
We need to consider the argument as annual. Currently, we treat it as 'daily'  (divide by DAY).

Reply:
OK I can rename the const to ALLOWED_BEACON_BALANCE_ANNUAL_INCREASE

Reply:
just a style nit: I suggest using CAPITAL_NAMES for constants. This is just a variable (for this contract). It should be `allowedBeaconBalanceIncrease` or something like that. 
But when you use unstructured storage (keccak256 offsets), they are constants: `ALLOWED_BEACON_BALANCE_INCREASE_POSITION` (note the POSITION in the end).

Reply:
Strictly speaking it's not increase but a relative increase